src/lib/RequestParser.js

```diff
@@ -192,4 +192,31 @@ class RequestParser { // eslint-disable-line no-unused-vars
             throw new Errors.InvalidRequestError(`Invalid url: ${error.message}`);
         }
     }
+
+    /**
+     * @param {any} url
+     * @returns {URL}
+     */
+    parseShopLogoUrl(url) {
+        if (!url || typeof url !== 'string') {
+            throw new Errors.InvalidRequestError('shopOrigin must be of type string');
```

Comment:
```suggestion
            throw new Errors.InvalidRequestError('shopLogoUrl must be of type string');
```

Reply:
`shopLogoUrl` is an optional parameter. This logic here does not treat it as optional but as required.